### ISSUE TITLE
Workflow Not Completing When Run From Self Service

### ISSUE THREAD

**Issue by dotdashhhh** [kc9wwh/macOSUpgrade#105]
**Testing from 10.11.6 to 10.13.6. Physical MacBook Pro (2012). Running policies from Self Service.**

The macOSUpgrade workflow seems broken when the script is initiated from Self Service. 

After the full screen warning dialogue (& from what I can discern from logs below... when startosinstall binary is called), the computer signs out of the account and sits at login screen doing nothing.

**Jamf logs report the normal stuff:**
`Script result: Power Check: OK - AC Power Detected<br/>Disk Check: OK - 463139450880 Bytes Free Space Detected<br/>/Applications/Install macOS High Sierra.app found, checking version.<br/>OSVersion is 10.13.6<br/>Installer found, version matches. Verifying checksum...<br/>Launching jamfHelper as FullScreen...<br/>/Library/LaunchAgents/com.apple.install.osinstallersetupd.plist: service already loaded<br/>Launching startosinstall...<br/>`

**Install.log did not have any useful info, at least that I could discern.**

**startosinstall.log provided the following after two attempts in Self Service:**
```
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.
```

When I ran the policy manually from the Terminal, the script ran successfully. I received the full screen warning dialogue. Then the computer restarted and is now upgrading to High Sierra.

It seems to me that when the policy is run from Self Service, is when the computer logs out the account and does not proceed as it should. 

Am I missing something or is this a bug? Having the users be able to choose when to perform their upgrade is critical for this being applicable in our environment. Would really like to be able to make use of this script.

Thank you for making this script and for providing any assistance/insight.

**Comment by kenchan0130** [comment]
Thank you for the detailed report.
I am able to understand.

> When I ran the policy manually from the Terminal, the script ran successfully. 

This is very useful information.
What command did you execute specifically?
Could it be that run `sudo jamf policy -event <triggerName>`?

**Comment by dotdashhhh** [comment]
@kenchan0130 

Correct. I use `sudo jamf policy -event <triggerName>` to execute the policy manually.

**Comment by kenchan0130** [comment]
@dotdashhhh 

Please tell me more about it.

1. Did you run command which is `sudo jamf policy -event <triggerName>` , the trigger name is like `download-high-sierra-install` ?
  - this means macOS installer download policy.
2. Or, did you run command which is `sudo jamf policy -id <policy id>`?
  -  this means macOSUpgrade policy.
